### ISSUE TITLE
IsKeyPressed or IO.KeysDown not working

### ISSUE THREAD

**Issue by LoicBaumann** [ocornut/imgui#6523]
Version/Branch of Dear ImGui:

Version: 1.89.2
Branch: Don't know

Back-end/Renderer/Compiler/OS
Using ImGuiNet / .net 7 / Windows

I feel this was something that used to work, I don't know if it's a change in my code or the recent upgrade to 1.89.2, but `ImGui.IsKeyPressed[551]` doesn't work anymore to detect a F key down.

I've tried to look in the source code of Dear ImGui for hint, I've tried `SetNextFrameWantCaptureKeyboard(true)`, also try to dump `GetIO.KeysData[]`. But I can't get a detection of a key down.

Editing text in an input text box is working, 
The Debug Log is outputing : `Processed: Text: f"

Basically, all I want is detecting if there's a given key pressed/released when a given Window has the focus.
I'm sure I'm missing something, but what?

Thanks

**Comment by ocornut** [comment]
You can browse Inputs in Metrics>Inputs and check IO in DebugLog>IO. 
It looks like your backend is submitting characters but not keys.

**Comment by LoicBaumann** [comment]
![image](https://github.com/ocornut/imgui/assets/48723305/53e3af70-47e1-401b-b7a1-4713d4bfd153)
Key Pressed displays the same thing that key down (obviously not when I take the screen shot).

By backend you mean the Windows implementation of DearImGui? I didn't change a thing, I remember upgrading my ImGuiNet package at some point, it also upgraded DearImGui IIRC.

Do you have pointers for me to try and troubleshoot/dig this further?

Thanks

**Comment by ocornut** [comment]
AFAIK ImGuiNet is a language binding for C#/.Net but not a backend itself, your problem seems to lie in your backend version.

**Comment by LoicBaumann** [comment]
Ok, I wasn't aware of the change and I think I'll find my answer in #4921 
Yes ImGuiNet is most about binding things to be used in .net.

```
Dear ImGui 1.89.2 (18920)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: IMGUI_DISABLE_OBSOLETE_FUNCTIONS
define: _WIN32
define: _WIN64
define: _MSC_VER=1934
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: NULL
io.BackendRendererName: NULL
io.ConfigFlags: 0x00000040
 DockingEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigWindowsMoveFromTitleBarOnly
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000A
 HasMouseCursors
 RendererHasVtxOffset
```


Thanks for the help.

**Comment by ocornut** [comment]
Find where you have calls to `io.AddInputCharacter()` and you'll find what/where your backend is, and then you need to look at updating this to use 1.87 IO API.

(In the meantime you can use `IsKeyPressed('F')` but that will be obsoleted and is backend-dependant so your code won't work with another backend.)

**Comment by LoicBaumann** [comment]
Ok, I found the call in ImGui.Net, I follow up there about the issue.
I'm closing this one, thanks again !

**Comment by ocornut** [comment]
Both https://github.com/ImGuiNET/ImGui.NET/blob/b5256ac381ceedbade8f41c3e287aa65f62d84ee/src/ImGui.NET.SampleProgram.XNA/ImGuiRenderer.cs and https://github.com/ImGuiNET/ImGui.NET/blob/b5256ac381ceedbade8f41c3e287aa65f62d84ee/src/ImGui.NET.SampleProgram/ImGuiController.cs seems to be up to date and behaving well so you must be using an old version.
